### PR TITLE
Changes to expand the usability of cache utils.

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**
- Support encrypted shared prefernces.
 - Determine the class of objects based on key and deserialize that way.

### CODE REVIEW

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/CacheUtils.java [hunk 1]
```diff
@@ -54,7 +62,7 @@ public String apply(String s) {
         }
     };
 
-    /**
+     /**
```
Comment (AdamBJohnsonx) [inline testutils/src/main/java/com/microsoft/identity/internal/testutils/CacheUtils.java:65]:
```suggestion
    /**
```

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java [hunk 2]
```diff
@@ -62,6 +74,190 @@ public static SharedPreferences getSharedPreferences(final String sharedPrefName
         return context.getSharedPreferences(sharedPrefName, Context.MODE_PRIVATE);
     }
 
+    /**
+     * Return a SharedPreferences instance that works with stores containing encrypted values.
+     * @param sharedPrefName the name of the shared preferences file.
+     * @return A SharedPreferences that decrypts and encrypts the values.
```
Comment (shahzaibj) [inline testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java:80]: needs formatting

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java [hunk 3]
```diff
@@ -62,6 +74,190 @@ public static SharedPreferences getSharedPreferences(final String sharedPrefName
...
+     */
+    public static SharedPreferences getEncryptedSharedPreferences(final String sharedPrefName) {
+        final Context context = ApplicationProvider.getApplicationContext();
+        final SharedPreferences barePreferences = context.getSharedPreferences(sharedPrefName, Context.MODE_PRIVATE);
+        final StorageHelper helper = new StorageHelper(context);
```
Comment (shahzaibj) [inline testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java:85]: nit: rename to `storageHelper`

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java (hunk 2 above)
Comment (AdamBJohnsonx) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java]: addressed.

File: testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java (hunk 3 above)
Comment (AdamBJohnsonx) [reply testutils/src/main/java/com/microsoft/identity/internal/testutils/TestUtils.java]: addressed.

**Review by shahzaibj — APPROVED** [review]

**Pull request merged**